clang-plugin/MozsearchIndexer.cpp

```diff
@@ -2534,17 +2534,27 @@ class IndexConsumer : public ASTConsumer,
   }
 
   void endMacroExpansion() {
-    const auto replacements = clang::format::reformat(
-      clang::format::getMozillaStyle(),
-      MacroExpansionState->Expansion,
-      {tooling::Range(0, MacroExpansionState->Expansion.length())}
-    );
-    auto formatted = clang::tooling::applyAllReplacements(MacroExpansionState->Expansion, replacements);
-    if (formatted) {
-      for (auto &[k, v] : MacroExpansionState->TokenLocations) {
-        v = replacements.getShiftedCodePosition(v);
-      }
-      MacroExpansionState->Expansion = std::move(formatted.get());
+    static constexpr auto includedFileExpansionReformatThreshold = 20'000;
+    static constexpr auto mainFileExpansionReformatThreshold = 200'000;
+
+    const auto expansionLocation = SM.getExpansionLoc(MacroExpansionState->MacroNameToken.getLocation());
+    const auto expansionFileId = SM.getFileID(expansionLocation);
+    const auto isIncludedFile = SM.getIncludeLoc(expansionFileId).isValid();
```

Comment:
It seems like this might get tricked by unified builds where the cpp files are `#include`d?

Reply:
Right, should I just filter by filename then?
I'm thinking of filtering out *.h, *.hpp, *.hxx and *.inc.

Reply:
Yeah, I think treating the most common include extensions as includes makes sense (given the horrible contortions used to make C++ builds fast).